nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -329,13 +335,15 @@ void DockingServer::dockRobot()
             stashDockData(goal->use_dock_id, dock, true);
             publishZeroVelocity();
             docking_action_server_->succeeded_current(result);
+            dock->plugin->stopDetectionProcess();
```

Comment:
Move this before `succeeded_current` line. More about consistency across the codebase than a technical issue. We tend to do the very last thing as succeeding / terminating actions so that timers that can be inserted for metrics